src/pages/experiments/[experimentId]/data-processing/StepsIndicator.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const StepsIndicator = (props) => {
+  const { allSteps, completedSteps, currentStep } = props;
+  let increment = 0;
+
+  return (
+    <>
+      {allSteps.map((step) => {
+        let color = '#1890ff';
```

Comment:
there is a more javascript-y way of doing this, which is instead of creating a variable "increment" outside of the `map` filter, you do it in the `map` filter like this:

`allSteps.map((step, increment) => {....})`